keras_cv/metrics/object_detection/box_coco_metrics.py

```diff
@@ -223,6 +223,13 @@ def name_prefix(self):
     def update_state(self, y_true, y_pred, sample_weight=None):
         self._eval_step_count += 1
 
+        if isinstance(y_true["boxes"], tf.RaggedTensor) != isinstance(
```

Comment:
@ianstenbit 

There was a subtle bug:
When inputs consist of a mixture of ragged and dense tensor, `tf.concat` in `_compute_result` will fail due to different sizes along the num_boxes dimension.
However, it works fine when the inputs are either all dense (with same shape) or all ragged.

This PR ensures that the tensor becomes ragged if any of the inputs are ragged.

I'm unsure if this is a good decision.

Reply:
This seems reasonable to me -- it might be more consistent to align on dense instead of ragged in this case, but ultimately I don't think it matters.